### PR TITLE
chore: update moby/docker, and runc for linux only

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR updates the default moby (docker) version to 20.10.7 for both Linux and Windows.

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: vhd/packer/configure-windows-vhd.ps1 [hunk 1]
```diff
@@ -176,7 +176,7 @@ function Install-ContainerD {
 }
 
 function Install-Docker {
-    $defaultDockerVersion = "20.10.5"
+    $defaultDockerVersion = "20.10.7"
```
Comment (jsturtevant) [inline vhd/packer/configure-windows-vhd.ps1:179]: Windows releases are via Mirantis Docker EE which can be behind the Linux moby releases and sometimes skip patch releases.  Currently the latest release version is `20.10.6`

File: pkg/api/const.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ const (
 
 const (
 	// KubernetesWindowsDockerVersion is the default version for docker on Windows nodes in kubernetes
-	KubernetesWindowsDockerVersion = "20.10.5"
+	KubernetesWindowsDockerVersion = "20.10.7"
```
Comment (jsturtevant) [inline pkg/api/const.go:34]: Same here for the Windows Docker EE version

**Review by jsturtevant — CHANGES_REQUESTED** [review]

File: vhd/packer/configure-windows-vhd.ps1 (hunk 1 above)
Comment (jsturtevant) [reply vhd/packer/configure-windows-vhd.ps1:179]: you can find latest from the link at: https://dockermsft.azureedge.net/dockercontainer/DockerMsftIndex.json

Comment (jackfrancis) [reply vhd/packer/configure-windows-vhd.ps1]: Got it, thanks!

File: pkg/engine/templates_generated.go [hunk 3]
```diff
@@ -16525,7 +16525,8 @@ var _k8sKubernetesparamsT = []byte(`    "etcdServerCertificate": {
          "19.03.12",
          "19.03.13",
          "19.03.14",
-         "20.10.5"
+         "20.10.5",
```
Comment (Michael-Sinz) [inline pkg/engine/templates_generated.go:16528]: Should this just be a value that is a default and then, if the user picks something else, the code just uses it?  As it is, I doubt that we have done regression testing against all of these versions and the current kuberenetes/aks-engine versions.

**Review by Michael-Sinz — APPROVED** [review]
Only question is why we have a list of allowed versions for moby/docker versions.  Are they somehow specially tested or just a history of what was once valid.

**Comment by jackfrancis** [comment]
> Only question is why we have a list of allowed versions for moby/docker versions. Are they somehow specially tested or just a history of what was once valid.

It's the latter. Just doing the minimal maintenance effort here, though it's worth acknowledging that making the effort to unwind the ARM template validations will make things less confusing. We can do that work in a discrete PR if we want.

**Comment by jackfrancis** [comment]
@jsturtevant @AbelHu updated Windows to use the .6 patch release of docker

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by AbelHu — APPROVED** [review]

**Review by jsturtevant — APPROVED** [review]

**Pull request merged**